bats/tests/helpers/defaults.bash

```diff
@@ -188,3 +193,47 @@ validate_enum RD_LOCATION system user dist dev ""
 using_dev_mode() {
     [ "$RD_LOCATION" = "dev" ]
 }
+
+########################################################################
+# RD_K3S_VERSIONS specifies a list of k3s versions. foreach_k3s_version()
+# can dynamically register a test to run once for each version in the
+# list. Only versions between RD_K3S_MIN and RD_K3S_MAX (inclusively)
+# will be used.
+#
+# Special values:
+# "all" will fetch the list of all k3s releases from GitHub
+# "latest" will fetch the list of latest versions from the release channel
+
+: "${RD_K3S_MIN:=1.0.0}"
+: "${RD_K3S_MAX:=1.99.0}"
+: "${RD_K3S_VERSIONS:=}"
+
+validate_semver RD_K3S_MIN
+validate_semver RD_K3S_MAX
+
+# Cache expansion of RD_K3S_VERSIONS special versions because they are slow to compute
+if ! load_var RD_K3S_VERSIONS; then
```

Comment:
This looks complicated enough that I think this should live in `kubernetes.bash` as a function (`get_k3s_versions` or something), so that reading this file is less confusing.

Reply:
I've done this for now, but it then requires `load.bash` to source `kubernetes.bash` before `defaults.bash`, which only works because nobody will ball `wait_for_kubelet` before the rest of the files have been sourced.

Reply:
Ah, sorry, I was unclear: I meant to replace the use of `RD_K3S_VERSIONS` with the function, not to do so inside `defaults.bash`. But I understand if you don't feel like changing this again…

Reply:
I see. But that means `all` and `latest` would be evaluated on first use, and not upfront. So if they fail for some reason (e.g. GitHub networking), they will run tests for 20min, and then break.

Or am I still misunderstanding you?

Reply:
Hmm, you're correct there; in that case they'll fail partway through the test run, instead of upfront.

I think that's the same as any other network error partway through the test (things fail and we have to dig into and realize that it's possibly intermittent).  Now that you know what I'm thinking, though, I'm happy with whichever path you think is better.